docs/reference/superjson.mdx

```diff
@@ -0,0 +1,294 @@
+{
+  "myusecase": {
+    "input": {
+      "someInputProperty": 42
+    },
+    "providerFailover": true
+  }
+}
+```
+
+### ProfileProviderSettings
+
+ProfileProviderSettings object is used for configuring specific providers for a profile. It can point to either file or specific variant and revision. It contains following optional property:
+
+- `defaults`: a record of defaults per usecase, key is usecase name and value is a [ProfileProviderDefaults](#profileproviderdefaults) object
+
+```json
+{
+  "file": "file:path/to/provider.json",
+  "defaults": {
+    "myusecase": {
+      // ...
+    }
+  }
+}
+```
+
+```json
+{
+  "mapVariant": "my-variant",
+  "mapRevision": "3",
+  "defaults": {
+    "myusecase": {
+      // ...
+    }
+  }
+}
+```
+
+### ProfileProviderDefaults
+
+ProfileProviderDefaults is an object with following optional properties:
+
+- `input`: an object providing default values according to usecase's input
+- `retryPolicy`: a [RetryPolicy](#retrypolicy) object
+
+```json
+{
+  "input": {
+    "someInputProperty": 42
+  },
+  "retryPolicy": {
+    // ...
+  }
+}
+```
+
+### RetryPolicy
+
+RetryPolicy is used for setting up retry policy for a provider and usecase. There are currently two types of retry policy, `none` or `circuit-breaker`. When using `none`, no retry is attempted. When using `circuit-breaker`, several attemps are made with backoff. Following properties are used to configure the policy (all optional except kind). When using default values, `none` or `circuit-breaker` string can be used as shorthand instead of the object.
+
+- `kind`: `none` or `circuit-breaker`
+- `maxContiguousRetries`: maximum number of retry attempts, positive integer, default 5
+- `requestTimeout`: timeout after which a request fails, positive integer, milliseconds, default 30000
+- `backoff`: either string `exponential` or an object with following properties:
+    - `kind`: string `exponential`
+    - `start`: the initial delay time, optional, positive integer, milliseconds, default 2000
+    - `factor`: the factor by which is the delay time multiplied, optional, positive integer, default 2
+
+```json
+{
+  "kind": "circuit-breaker",
+  "maxContiguousRetries": 3,
+  "requestTimeout": 60000,
+  "backoff": {
+    "kind": "exponential",
+    "start": 1000,
+    "factor": 3
+  }
+}
+```
+
+## Providers
+
+The providers section consists of a record of all used providers. The key is the name of the provider and the value is either a file URI or a [ProviderSettings](#providersettings) object.
+
+```json
+{
+  "providers": {
+    "provider1": "file:path/to/provider.json",
+    "provider2": {
+      // ...
+    }
+  }
+}
+```
+
+### ProviderSettings
+
+The ProviderSettings object has following optional properties:
+
+- `file`: a file URI string
+- `security`: an array of [SecurityValues](#securityvalues) objects
+
+```json
+{
+  "file": "file:path/to/provider.json",
+  "security": [
+      {
+        // ...
+      }
+  ]
+}
+```
+
+### SecurityValues
```

Comment:
I don't see any explanation for using `$` for injecting env variables. Can it be used anywhere or only for security values?